api/api-abstract-material-test/src/main/java/com/thoughtworks/go/api/abstractmaterialtest/AbstractMaterialTestController.java

```diff
@@ -126,12 +123,18 @@ private String handleValidationBeanResponse(ValidationBean validationBean, Respo
     }
 
     private void performParamExpansion(ScmMaterialConfig scmMaterialConfig, String pipelineName) {
-        PipelineConfig existingPipeline = goConfigService.pipelineConfigNamed(new CaseInsensitiveString(pipelineName));
-        PipelineConfig pipelineConfig = new PipelineConfig(existingPipeline.name(), new MaterialConfigs());
-
-        GoConfigCloner goConfigCloner = new GoConfigCloner();
-        pipelineConfig.setParams(goConfigCloner.deepClone(existingPipeline.getParams()));
+        PipelineConfig pipelineConfig;
+        // If the pipeline name is provided, find the pipeline and add the params to the new pipeline config object
+        if (isNotBlank(pipelineName)) {
+            PipelineConfig existingPipeline = goConfigService.pipelineConfigNamed(new CaseInsensitiveString(pipelineName));
+            pipelineConfig = new PipelineConfig(existingPipeline.name(), new MaterialConfigs());
 
+            GoConfigCloner goConfigCloner = new GoConfigCloner();
+            pipelineConfig.setParams(goConfigCloner.deepClone(existingPipeline.getParams()));
+        } else {
+            // If the pipeline name is not provided, this means that the pipeline is still in creation nd hence no params exist
```

Comment:
@kritika-singh3 do you think we can add a test for these scenarios over adding a comment?